lib/util/version.js

```diff
@@ -12,7 +13,23 @@ function getReactVersionFromContext(context) {
   let confVer = '999.999.999';
   // .eslintrc shared settings (http://eslint.org/docs/user-guide/configuring#adding-shared-settings)
   if (context.settings.react && context.settings.react.version) {
-    confVer = context.settings.react.version;
+    let settingsVer = context.settings.react.version;
+    if (settingsVer === 'detect') {
+      try {
+        const reactPath = resolve.sync('react', {basedir: process.cwd()});
+        const react = require(reactPath);
+        settingsVer = react.version;
```

Comment:
@ljharb does it make sense to cache this value for subsequent access?

Reply:
yes, absolutely - good call.